cpu/stm32/periph/spi.c

```diff
@@ -256,8 +256,9 @@ static void _transfer_dma(spi_t bus, const void *out, void *in, size_t len)
     dma_wait(spi_config[bus].rx_dma);
     dma_wait(spi_config[bus].tx_dma);
 
-    /* No need to stop the DMA here, it is automatically disabled when the
-     * transfer is finished, only wait for SPI to leave the busy state */
+    dma_stop(spi_config[bus].rx_dma);
+    dma_stop(spi_config[bus].tx_dma);
```

Comment:
@bergzand you worked with DMA optimization a bunch I think, is this worth optimizing for platforms who do or do not need to use the stop?

Reply:
I think it is worth optimizing it a bit, but definitely out of scope here. I agree with bringing back `dma_stop()` here.

Reply:
Test on f746zg without dma_stop:
```
2020-08-18 13:53:08,734 #  init 0 0 4 0 0
2020-08-18 13:53:08,740 # SPI_DEV(0) initialized: mode: 0, clk: 4, cs_port: 0, cs_pin: 0
bench
2020-08-18 13:53:14,653 #  bench
2020-08-18 13:53:14,658 # ### Running some benchmarks, all values in [us] ###
2020-08-18 13:53:14,661 # ### Test				Transfer time	user time
2020-08-18 13:53:14,661 # 
2020-08-18 13:53:14,672 #  1 - write 1000 times 1 byte:			7159	7162
2020-08-18 13:53:14,688 #  2 - write 1000 times 2 byte:			11913	11915
2020-08-18 13:53:14,821 #  3 - write 1000 times 100 byte:		129408	9782
2020-08-18 13:53:14,841 #  4 - write 1000 times 1 byte to register:	14693	14695
2020-08-18 13:53:14,865 #  5 - write 1000 times 2 byte to register:	19945	19949
2020-08-18 13:53:15,007 #  6 - write 1000 times 100 byte to register:	136951	17615
2020-08-18 13:53:15,023 #  7 - read 1000 times 2 byte:			11962	11964
2020-08-18 13:53:15,156 #  8 - read 1000 times 100 byte:		129075	9449
2020-08-18 13:53:15,181 #  9 - read 1000 times 2 byte from register:	20302	20305
2020-08-18 13:53:15,323 # 10 - read 1000 times 100 byte from register:	137127	17879
2020-08-18 13:53:15,339 # 11 - transfer 1000 times 2 byte:		11762	11764
2020-08-18 13:53:15,473 # 12 - transfer 1000 times 100 byte:		129093	9484
2020-08-18 13:53:15,498 # 13 - transfer 1000 times 2 byte to register:	19927	19930
2020-08-18 13:53:15,640 # 14 - transfer 1000 times 100 byte to register:136965	17634
2020-08-18 13:53:15,654 # 15 - acquire/release 1000 times:		10081	10084
2020-08-18 13:53:16,657 # -- - SUM:					926363	209611
2020-08-18 13:53:16,657 # 
2020-08-18 13:53:16,659 # ### All runs complete ###
```
Then with dma_stop:
```
2020-08-18 13:54:02,402 # init 0 0 4 0 0
2020-08-18 13:54:02,407 # SPI_DEV(0) initialized: mode: 0, clk: 4, cs_port: 0, cs_pin: 0
bench
2020-08-18 13:54:11,806 #  bench
2020-08-18 13:54:11,811 # ### Running some benchmarks, all values in [us] ###
2020-08-18 13:54:11,814 # ### Test				Transfer time	user time
2020-08-18 13:54:11,814 # 
2020-08-18 13:54:11,826 #  1 - write 1000 times 1 byte:			8447	8449
2020-08-18 13:54:11,843 #  2 - write 1000 times 2 byte:			12743	12746
2020-08-18 13:54:11,976 #  3 - write 1000 times 100 byte:		130242	10707
2020-08-18 13:54:11,998 #  4 - write 1000 times 1 byte to register:	16589	16592
2020-08-18 13:54:12,024 #  5 - write 1000 times 2 byte to register:	22131	22133
2020-08-18 13:54:12,169 #  6 - write 1000 times 100 byte to register:	138918	19422
2020-08-18 13:54:12,186 #  7 - read 1000 times 2 byte:			13322	13326
2020-08-18 13:54:12,320 #  8 - read 1000 times 100 byte:		129964	10448
2020-08-18 13:54:12,347 #  9 - read 1000 times 2 byte from register:	22408	22411
2020-08-18 13:54:12,492 # 10 - read 1000 times 100 byte from register:	139159	19643
2020-08-18 13:54:12,508 # 11 - transfer 1000 times 2 byte:		12638	12641
2020-08-18 13:54:12,644 # 12 - transfer 1000 times 100 byte:		129992	10435
2020-08-18 13:54:12,669 # 13 - transfer 1000 times 2 byte to register:	22103	22106
2020-08-18 13:54:12,814 # 14 - transfer 1000 times 100 byte to register:138937	19410
2020-08-18 13:54:12,827 # 15 - acquire/release 1000 times:		10067	10070
2020-08-18 13:54:13,831 # -- - SUM:					947660	230539
2020-08-18 13:54:13,831 # 
2020-08-18 13:54:13,833 # ### All runs complete ###
```
The impact seems significant to me so I have investigated further.

The EN bit description of the devices with SxCR register says that `This bit may be cleared by hardware:
–
on a DMA end of transfer (stream ready to be configured)` unlike those with CCR register so I think that only those with DMA_CCR_EN defined need those lines, not those with DMA_SxCR_EN.

Reply:
@bergzand Your opinion ?